src/main/java/com/jit/studentmanagementsystem/controllers/TeacherController.java

```diff
@@ -0,0 +1,55 @@
+package com.jit.studentmanagementsystem.controllers;
+
+import com.jit.studentmanagementsystem.models.Teacher;
+import com.jit.studentmanagementsystem.repositories.TeacherRepository;
+import org.springframework.beans.BeanUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/v1/teachers")
+public class TeacherController {
+    @Autowired
+    private TeacherRepository teacherRepository;
+
+    @GetMapping
+    public List<Teacher> list() {
+        return teacherRepository.findAll();
+    }
+
+    @GetMapping("{id}")
+    public Teacher get(@PathVariable Long id) {
+        return teacherRepository.getOne(id);
+    }
+
+    @PostMapping
+    public Teacher create(@RequestBody final Teacher teacher) {
+        return teacherRepository.saveAndFlush(teacher);
+    }
+
+    @RequestMapping(value = "{id}", method = RequestMethod.DELETE)
+    public void delete(@PathVariable Long id) {
+        teacherRepository.deleteById(id);
+    }
+
+    @RequestMapping(value = "{id}", method = RequestMethod.PUT)
+    public ResponseEntity update(@PathVariable Long id, @RequestBody Teacher teacher) {
```

Comment:
why not service for that?

Reply:
I don't understand what you mean please.

Reply:
this is only palace when you call repository directly in controller without service

Reply:
as in this controller TeacherController vs StudentController

Reply:
Ok, I now see clearly. I am yet to make proper changes on the Teacher and Student controllers. I just focused on the Course controller since it's implementation will affect I implement others. I will soon work on them.